src/datasets/iterable_dataset.py

```diff
@@ -1115,6 +1116,24 @@ def from_spark(
             **kwargs,
         ).read()
 
+    @staticmethod
+    def from_file(filename: str, in_memory: bool = False) -> "IterableDataset":
+        """Instantiate a IterableDataset from Arrow table at filename.
+
+        Args:
+            filename (`str`):
+                File name of the dataset.
+            in_memory (`bool`, defaults to `False`):
+                Whether to copy the data in-memory.
+
+        Returns:
+            [`IterableDataset`]
+        """
+        pa_table = ArrowReader.read_table(filename, in_memory=in_memory)
```

Comment:
This would read the entire file metadata, which can take some time, in particular if you have thousands of files (which is going to be the case for people interested in https://github.com/huggingface/datasets/issues/5481)

I think it should be possible to have an arrow table generator that yields tables from the file without reading all its metadata.

Maybe using `pyarrow.ipc.RecordBatchStreamReader` and then `read_next_batch` ?